examples/error-handling/src/main.rs

```diff
@@ -0,0 +1,232 @@
+//! Example showing how to convert errors into responses.
+//!
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p example-error-handling
+//! ```
+//!
+//! For successful requests the log output will be
+//!
+//! ```ignore
+//! DEBUG request{method=POST uri=/users matched_path="/users"}: tower_http::trace::on_request: started processing request
+//! DEBUG request{method=POST uri=/users matched_path="/users"}: tower_http::trace::on_response: finished processing request latency=0 ms status=200
+//! ```
+//!
+//! For failed requests the log output will be
+//!
+//! ```ignore
+//! DEBUG request{method=POST uri=/users matched_path="/users"}: tower_http::trace::on_request: started processing request
+//! ERROR request{method=POST uri=/users matched_path="/users"}: example_error_handling: error from time_library err=failed to get time
+//! DEBUG request{method=POST uri=/users matched_path="/users"}: tower_http::trace::on_response: finished processing request latency=0 ms status=500
+//! ```
+
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, Mutex,
+    },
+};
+
+use axum::{
+    extract::{rejection::JsonRejection, FromRequest, MatchedPath, Request, State},
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    routing::post,
+    Router,
+};
+use serde::{Deserialize, Serialize};
+use time_library::Timestamp;
+use tower_http::trace::TraceLayer;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::registry()
+        .with(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| "example_error_handling=debug,tower_http=debug".into()),
+        )
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    let state = AppState::default();
+
+    let app = Router::new()
+        // A dummy route that accepts some JSON but sometimes fails
+        .route("/users", post(users_create))
+        .layer(
+            TraceLayer::new_for_http()
+                // Create our own span for the request and include the matched path. The matched
+                // path is useful for figuring out which handler the request was routed to.
+                .make_span_with(|req: &Request| {
+                    let method = req.method();
+                    let uri = req.uri();
+
+                    // axum automatically adds this extension.
+                    let matched_path = req
+                        .extensions()
+                        .get::<MatchedPath>()
+                        .map(|matched_path| matched_path.as_str());
+
+                    tracing::debug_span!("request", %method, %uri, matched_path)
```

Comment:
As-is, the uri (and method) won't be in quotes but the matched path will, as the comment at the start of the file shows correctly. I find this pretty weird. Maybe use `let uri = req.uri().to_string();` above instead, and remove the `%` before `uri` here? For the method, it probably makes sense for it not to be in quotes.